core/modules/field_ui/field_ui.admin.inc

```diff
@@ -45,26 +46,39 @@ function field_ui_fields_list() {
           $rows[$field_name]['class'] = $field['locked'] ? array('menu-disabled') : array('');
           $rows[$field_name]['data'][0] = $field['locked'] ? t('@field_name (Locked)', array('@field_name' => $field_name)) : $field_name;
           $module_name = $field_types[$field['type']]['module'];
-          $rows[$field_name]['data'][1] = $field_types[$field['type']]['label'] . ' ' . t('(module: !module)', array('!module' => $modules[$module_name]->info['name']));
+          $rows[$field_name]['data'][1] = $field_types[$field['type']]['label'];
+          $rows[$field_name]['data'][2] = $modules[$module_name]->info['name'];
         }
 
         // Add the current instance.
         $admin_path = _field_ui_bundle_admin_path($entity_type, $bundle);
-        $rows[$field_name]['data'][2][] = $admin_path ? l($bundles[$entity_type][$bundle]['label'], $admin_path . '/fields') : $bundles[$entity_type][$bundle]['label'];
+        $field_label = $instances[$entity_type][$bundle][$field_name]['label'];
+        $bundle_label = $bundles[$entity_type][$bundle]['label'];
+        if ($admin_path) {
+          $bundle_manage_fields_link = $admin_path . '/fields';
+          $field_configure_link = $bundle_manage_fields_link . '/' . $field_name;
+          $rows[$field_name]['data'][3][] = t('<a href="@field_configure_link">%field_label</a> in <a href="@bundle_manage_fields_link">@bundle_label</a>', array('@field_configure_link' => url($field_configure_link), '%field_label' => $field_label, '@bundle_manage_fields_link' => url($bundle_manage_fields_link), '@bundle_label' => $bundle_label));
+        }
+        else {
+          $rows[$field_name]['data'][3][] = t('%field_label in @bundle_label', array('%field_label' => $field_label, '@bundle_label' => $bundle_label));
+        }
       }
     }
   }
   foreach ($rows as $field_name => $cell) {
-    $rows[$field_name]['data'][2] = implode(', ', $cell['data'][2]);
+    // Render the items of the "Used in" column as a list.
+    $rows[$field_name]['data'][3] = theme('item_list', array('items' => $rows[$field_name]['data'][3]));
```

Comment:
As explained in the issue, a reused filed can have different label across the various bundles it is being used in.